atum/src/main/scala/za/co/absa/atum/utils/controlmeasure/ControlMeasureBuilder.scala

```diff
@@ -52,28 +78,29 @@ object ControlMeasureBuilder {
   def forDF(df: DataFrame): ControlMeasureBuilder =
     ControlMeasureBuilderImpl(df)
 
-
   /**
    * This class can be used to construct a [[ControlMeasure]] data object (to be used for content as source _INFO file)
    * for a given dataframe using the `build` "method" after all necessary fields have been set.
    *
    * The row count measurement is added automatically. You can also specify aggregation columns for aggregation measurements
    *
-   * @param df                    A dataframe for which _INFO file to be created.
-   * @param aggregateColumns      Column names for `df`.
-   * @param sourceApplication     The name of the application providing the data (default = "").
-   * @param inputPathName         The path to the input file name. Can be a folder with file mask (default = "").
-   * @param reportDate            The date of the data generation (default = today).
-   * @param reportVersion         The version of the data generation for the date, new versions replace old  versions of data (default = 1).
-   * @param country               Country name (default = "ZA").
-   * @param historyType           History type (default = "Snapshot").
-   * @param sourceType            Source type (default = "Source").
-   * @param initialCheckpointName The name of the initial checkpoint (default = "Source").
-   * @param workflowName          A workflow name to group several checkpoint sth in the chain (default = "Source").
+   * @param df                        A dataframe for which _INFO file to be created.
+   * @param aggregateColumnMappings   Column names for `df` and their ControlTypeMappings (Default or Specific for each col)
+   * @param sourceApplication         The name of the application providing the data (default = "").
+   * @param inputPathName             The path to the input file name. Can be a folder with file mask (default = "").
+   * @param reportDate                The date of the data generation (default = today).
+   * @param reportVersion             The version of the data generation for the date, new versions replace old  versions of data (default = 1).
+   * @param country                   Country name (default = "ZA").
+   * @param historyType               History type (default = "Snapshot").
+   * @param sourceType                Source type (default = "Source").
+   * @param initialCheckpointName     The name of the initial checkpoint (default = "Source").
+   * @param workflowName              A workflow name to group several checkpoint sth in the chain (default = "Source").
    *
    */
   private case class ControlMeasureBuilderImpl(df: DataFrame,
-                                               aggregateColumns: Seq[String] = Seq.empty,
+                                               aggregateColumnMappings: Seq[ControlTypeMapping] = Seq.empty,
+                                               aggregateControlTypeStrategy: ControlTypeStrategy
```

Comment:
Missing in the docs above and I don't see it used

Reply:
Yes, that was one of the old implementation versions, will remove it.

Reply:
Removed.